xds/src/test/java/io/grpc/xds/XdsSdsClientServerTest.java

```diff
@@ -181,13 +203,21 @@ private static int findFreePort() throws IOException {
   }
 
   private SimpleServiceGrpc.SimpleServiceBlockingStub getBlockingStub(
-      UpstreamTlsContext upstreamTlsContext, String overrideAuthority) {
-    XdsChannelBuilder builder =
-        XdsChannelBuilder.forTarget("localhost:" + port).tlsContext(upstreamTlsContext);
+      final UpstreamTlsContext upstreamTlsContext, String overrideAuthority) {
     if (overrideAuthority != null) {
-      builder = builder.overrideAuthority(overrideAuthority);
+      channelBuilder = channelBuilder.overrideAuthority(overrideAuthority);
     }
-    return SimpleServiceGrpc.newBlockingStub(cleanupRule.register(builder.build()));
+    InetSocketAddress socketAddress =
+        new InetSocketAddress(Inet4Address.getLoopbackAddress(), port);
+    Attributes attrs =
+        (upstreamTlsContext != null)
+            ? Attributes.newBuilder()
+                .set(XdsAttributes.ATTR_UPSTREAM_TLS_CONTEXT, upstreamTlsContext)
+                .build()
+            : Attributes.EMPTY;
+    fakeNameResolverFactory.setServers(
+        Collections.singletonList(new EquivalentAddressGroup(socketAddress, attrs)));
```

Comment:
nit/java optional suggestion: using ImmutableList is preferred than Collections.singletonList especially if ImmutableList is already imported.